Mathlib/Data/List/Chain.lean

```diff
@@ -419,6 +419,30 @@ lemma Chain'.iterate_eq_of_apply_eq {α : Type*} {f : α → α} {l : List α}
     apply hl
     omega
 
+theorem chain'_replicate (n : ℕ) (a : α) : Chain' (· = ·) (replicate n a) :=
+  match n with
+  | 0 => chain'_nil
+  | 1 => chain'_singleton _
+  | n + 2 => by
+    rw [replicate_succ, replicate_succ, chain'_cons]
+    exact ⟨rfl, chain'_replicate (n + 1) a⟩
+
+theorem chain'_eq_iff_eq_replicate {l : List α} :
+    Chain' (· = ·) l ↔ ∀ a ∈ l.head?, l = replicate l.length a :=
```

Comment:
Does `eq_replicate_length` help you at all in this proof?

Reply:
It might help, but it would probably require more imports in order to use `List.chain.pairwise`. The proof would still have to separate out the case of an empty list, so it wouldn't even be much shorter.

Reply:
I think we might be missing some API. Ideally the proof would use that `Chain'` iff `Pairwise` (for a transitive relation), then `Pairwise` iff all elements are related (for a symmetric and reflexive relation). We have the former, but not the latter.